### ISSUE TITLE
Fulltextseach reindex is deleting too many pages when fluent is used

### ISSUE THREAD

**Issue by AljosaB** [tractorcow-farm/silverstripe-fluent#501]
When I run fulltextsearch reindex, existing records gets removed from Solr before new are being added (SolrReindexBase::clearRecords is being called in SolrReindexBase::runGroup). clearRecords than calls FluentSearchVariant::alterQuery to get the language part of delete.

```
public function alterQuery($query, $index)
{
	if (FluentState::singleton()->getIsFrontend() && Locale::getCached()->count()) {
		...
	}
}
```

But alterQuery never really gets applied because reindex is either run as queued job or from console (immediate reindex takes some time).

So if I have 5 languages, before it starts indexing second one it removes all already indexed pages and so on, and I end up having only the 5th one indexed.

clearRecord conditions:
```
array (
  0 => '+(ClassHierarchy:Page)',
  1 => '+(_subsite:"0" (*:* -_subsite:[* TO *]))',
  2 => '+(_versionedstage:"Stage" (*:* -_versionedstage:[* TO *]))',
)
```

clearRecord conditions with IF removed from alterQuery:
```
array (
  0 => '+(ClassHierarchy:Page)',
  1 => '+(_locale:"en_US" (*:* -_locale:[* TO *]))',
  2 => '+(_subsite:"0" (*:* -_subsite:[* TO *]))',
  3 => '+(_versionedstage:"Stage" (*:* -_versionedstage:[* TO *]))',
)
```

**Comment by robbieaverill** [comment]
Yeah that "is frontend" logic seems a bit odd. We shouldn't be assuming people will run reindex tasks from the frontend (browser initiated) - it's far more likely that it would be run via CLI or as a queued job.
